quinn-proto/src/connection/assembler.rs

```diff
@@ -144,16 +226,21 @@ impl Assembler {
         if bytes.is_empty() {
```

Comment:
We should maybe just move this all the way up

Reply:
Yeah, thought about that too.

Reply:
I will do that in a follow-up, otherwise the diff will be harder to review.

Reply:
Is it? I really meant **all** the way, as one of the first things in `insert`.  Hmm, or does it need to be here to catch the case where we have nothing after removing duplicate data?

Reply:
Oops, for some reason I thought you were talking about moving `insert` as a whole up, my bad. But yeah, it is there because `bytes` could become empty after deduplication.